File: go-sdk/pkg/models/shared/taskstripe.go
package shared

import (
	"time"
)

type TaskStripeDescriptor struct {
	Account string `json:"account"`
	Main    *bool  `json:"main,omitempty"`
	Name    string `json:"name"`
}

type TaskStripe struct {
	ConnectorID string                 `json:"connectorId"`
	CreatedAt   time.Time              `json:"createdAt"`
	Descriptor  TaskStripeDescriptor   `json:"descriptor"`
	Error       *string                `json:"error,omitempty"`
	ID          string                 `json:"id"`
	State       map[string]interface{} `json:"state"`
	Status      PaymentStatusEnum      `json:"status"`
	UpdatedAt   time.Time              `json:"updatedAt"`
}

File: ts-sdk/src/sdk/models/shared/taskstripe.ts
import { SpeakeasyBase, SpeakeasyMetadata } from "../../../internal/utils";
import { PaymentStatusEnum } from "./paymentstatusenum";


export class TaskStripeDescriptor extends SpeakeasyBase {
  @SpeakeasyMetadata({ data: "json, name=account" })
  account: string;

  @SpeakeasyMetadata({ data: "json, name=main" })
  main?: boolean;

  @SpeakeasyMetadata({ data: "json, name=name" })
  name: string;
}

export class TaskStripe extends SpeakeasyBase {
  @SpeakeasyMetadata({ data: "json, name=connectorId" })
  connectorId: string;

  @SpeakeasyMetadata({ data: "json, name=createdAt" })
  createdAt: Date;

  @SpeakeasyMetadata({ data: "json, name=descriptor" })
  descriptor: TaskStripeDescriptor;

  @SpeakeasyMetadata({ data: "json, name=error" })
  error?: string;

  @SpeakeasyMetadata({ data: "json, name=id" })
  id: string;

  @SpeakeasyMetadata({ data: "json, name=state" })
  state: Record<string, any>;

  @SpeakeasyMetadata({ data: "json, name=status" })
  status: PaymentStatusEnum;

  @SpeakeasyMetadata({ data: "json, name=updatedAt" })
  updatedAt: Date;
}